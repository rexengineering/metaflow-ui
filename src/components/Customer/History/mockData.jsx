import { faCommentAltLines } from "@fortawesome/pro-light-svg-icons/faCommentAltLines";
import { faPhoneAlt } from "@fortawesome/pro-light-svg-icons/faPhoneAlt";
import { faCommentAlt } from "@fortawesome/pro-regular-svg-icons/faCommentAlt";

export const CustomerHistory = [
  {
    label: "Call - Connected",
    date: "12/20/19",
    icon: faPhoneAlt,
    color: "#FBAE41",
  },
  {
    label: "Chat - Connected",
    date: "12/20/18",
    icon: faCommentAlt,
    color: "#807DFB",
  },
  {
    label: "Chat - Connected",
    date: "12/20/21",
    icon: faCommentAltLines,
    color: "#FF685F",
  },
];

export const Months = [
  "December 2020",
  "November 2020",
  "October 2020",
  "September 2020",
];
