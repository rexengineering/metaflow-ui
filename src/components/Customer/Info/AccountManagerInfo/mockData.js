const data = {
  name: "Duo Maxwell",
  role: "Account Manager",
  phoneNumber: "555-555-5555",
  address: "123 Main Street, Los Angeles CA 90094",
  onTransferCallClick: () => {},
};

export default data;
