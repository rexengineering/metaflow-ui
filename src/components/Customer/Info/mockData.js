export const MockedClientInfo = {
  name: "John Doe",
  email: "jdoe@email.com",
  phone: "3105555555",
  mobile: "3105555555",
  type: "Two",
};

export const MockedUserTypes = [
  {
    value: "One",
    label: "One",
  },
  {
    value: "Two",
    label: "Two",
  },
  {
    value: "Three",
    label: "Three",
  },
];
