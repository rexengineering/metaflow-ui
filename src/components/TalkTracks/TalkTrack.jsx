import React, {useEffect} from "react";
import {
  IconButton,
  makeStyles,
  Tab,
  Tabs,
  Typography,
} from "@material-ui/core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPlus } from "@fortawesome/pro-solid-svg-icons/faPlus";
import PropTypes from "prop-types";
import { talkTrackItemShape } from "../../utils/shapes";
import ActionCard from "../ActionCard";
import TalkTrackItem from "./TalkTrackItem";
import TabPanel from "../TabPanel";

const useStyles = makeStyles((theme) => ({
  header: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    position: "relative",
  },
  addButton: {
    position: "absolute",
    top: 0,
    right: 0,
    color: theme.palette.common.black,
  },
  talkTracks: {
    marginTop: theme.spacing(5),
    display: "flex",
  },
  tabs: {
    borderRight: `1px solid ${theme.palette.divider}`,
    width: "50%",
    flexGrow: 1,
  },
  tabsPanel: {
    width: "100%",
    flexGrow: 1,
  },
  tabsIndicator: {
    backgroundColor: theme.palette.primary.main,
  },
  title: {
    marginTop: theme.spacing(1),
  }
}));

function TalkTrack({ talkTrackItems, activeTalkTrackID, onSkip, onActionSelected, onTabChange, onContinue, className }) {
  const classes = useStyles();
  const [value, setValue] = React.useState(activeTalkTrackID);
  const handleChange = (event, newValue) => {
    onTabChange(newValue);
    setValue(newValue);
  };

  useEffect(() => {
    setValue(activeTalkTrackID);
  }, [activeTalkTrackID, setValue])

  return (
    <ActionCard className={className}>
      <section className={classes.header}>
        <Typography variant="h4" className={classes.title}>Talk track</Typography>
        <IconButton type="button" color="default"  className={classes.addButton}>
          <FontAwesomeIcon icon={faPlus} />
        </IconButton>
      </section>
      <section className={classes.talkTracks}>
        <Tabs
          classes={{ indicator: classes.tabsIndicator }}
          className={classes.tabs}
          onChange={handleChange}
          orientation="vertical"
          value={value}
        >
          {Array.isArray(talkTrackItems) &&
            talkTrackItems.map(({ identifier, title }) => (
              <Tab
                key={identifier}
                label={title}
                value={identifier}
              />
            ))}
        </Tabs>
        <section className={classes.tabsPanel}>
          {Array.isArray(talkTrackItems) &&
            talkTrackItems.map(
              ({
                identifier,
                steps,
                currentStep
              }) => (
                <TabPanel
                  key={identifier}
                  index={`${value}`}
                  value={identifier}
                >
                  {
                    steps.map(({order, title, speech, actions, active}) =>
                      order === currentStep
                      ? (
                            <TalkTrackItem
                                currentStep={currentStep}
                                key={order}
                                title={title}
                                speech={speech}
                                actions={actions}
                                onSkip={(stepIdentifier) => onSkip({stepIdentifier, talkTrackIdentifier: identifier})}
                                identifier={`${order}`}
                                onActionSelected={onActionSelected}
                                active={active}
                                onContinue={(stepIdentifier) => onContinue({stepIdentifier, talkTrackIdentifier: identifier})}
                            />
                          )
                      : null)
                  }
                </TabPanel>
              )
            )}
        </section>
      </section>
    </ActionCard>
  );
}

TalkTrack.defaultProps = {
  className: "",
}

TalkTrack.propTypes = {
  talkTrackItems: PropTypes.arrayOf(PropTypes.shape(talkTrackItemShape))
    .isRequired,
  className: PropTypes.string,
  activeTalkTrackID: PropTypes.string.isRequired,
  onSkip: PropTypes.func.isRequired,
  onActionSelected: PropTypes.func.isRequired,
  onContinue: PropTypes.func.isRequired,
  onTabChange: PropTypes.func.isRequired,
};

export default TalkTrack;
