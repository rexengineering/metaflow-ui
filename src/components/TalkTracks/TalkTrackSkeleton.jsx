import {makeStyles} from "@material-ui/core";
import {Skeleton} from "@material-ui/lab";

const useStyles = makeStyles(({spacing}) => ({
    tabs: {
        display: "flex",
        backgroundColor: "whitesmoke",
        padding: spacing(2),
    },
    options:{
        display: "flex",
    },
    content: {
        flexGrow: 2,
    },
    tabSeparator: {
        margin: spacing(0, 3),
        transform: "none"
    },
    contentHeader:{
        display: "flex",
        alignItems: "center",
        marginBottom: spacing(1),
    },
    contentHeaderText: {
        marginLeft: spacing(1),
        flexGrow: 1,
    },
}));

function TalkTrackSkeleton(){
    const classes = useStyles();
    const tabsSeparatorWidth = 3;
    const tabsSeparatorHeight = 200;
    const tabOptionWidth = 100;
    const tabOptionHeight = 50;
    const textHeight = 150;
    const headerCircleSize = 40;

    return (
        <section className={classes.tabs}>
            <section className={classes.options}>
                <div>
                    <Skeleton width={tabOptionWidth} height={tabOptionHeight} />
                    <Skeleton width={tabOptionWidth} height={tabOptionHeight} />
                    <Skeleton width={tabOptionWidth} height={tabOptionHeight} />
                    <Skeleton width={tabOptionWidth} height={tabOptionHeight} />
                </div>
                <Skeleton
                    className={classes.tabSeparator}
                    width={tabsSeparatorWidth}
                    height={tabsSeparatorHeight} />
            </section>
            <section className={classes.content}>
                <div className={classes.contentHeader}>
                    <Skeleton variant="circle" width={headerCircleSize} height={headerCircleSize} />
                    <div className={classes.contentHeaderText}>
                        <Skeleton />
                        <Skeleton />
                    </div>
                </div>
                <Skeleton variant="rect" height={textHeight} />
            </section>
        </section>
    );

}

export default TalkTrackSkeleton;