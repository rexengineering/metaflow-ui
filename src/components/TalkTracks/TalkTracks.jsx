import React from "react";
import PropTypes from "prop-types";
import {
  Card, CardContent, CardHeader, makeStyles, Typography,
} from "@material-ui/core";
import Workflow from "../Workflow";

const useStyles = makeStyles((theme) => ({
  workflow: {
    marginTop: theme.spacing(1),
    background: theme.palette.background.paper,
    padding: theme.spacing(2),
  },
}));

function TalkTracks({ talkTrackWorkflows, headerAction, instanceID }) {
  const classes = useStyles();

  return (
    <Card>
      <CardHeader title="Talk Tracks" action={headerAction} />
      <CardContent>
        {(!Array.isArray(talkTrackWorkflows) || !talkTrackWorkflows.length) && (
          <Typography>There are no active talk tracks.</Typography>
        )}
        {Array.isArray(talkTrackWorkflows) &&
          talkTrackWorkflows.map((workflowID) => (
            <Workflow
              key={workflowID}
              className={classes.workflow}
              workflowID={workflowID}
              instanceID={instanceID}
            />
        ))}
      </CardContent>
    </Card>
  );
}

TalkTracks.propTypes = {
  talkTrackWorkflows: PropTypes.arrayOf(PropTypes.string).isRequired,
  headerAction: PropTypes.node,
  instanceID: PropTypes.string.isRequired,
};

TalkTracks.defaultProps = {
  headerAction: undefined,
};

export default TalkTracks;
