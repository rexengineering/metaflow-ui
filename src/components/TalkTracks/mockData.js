const talkTracks = [
    {
        iid: "buying-123",
        isTalkTrack: true,
    },
    {
        iid: "selling-123",
        isTalkTrack: true,
    },
];

export default talkTracks;