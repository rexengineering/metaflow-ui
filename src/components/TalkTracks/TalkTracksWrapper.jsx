import React from "react";
import PropTypes from "prop-types";
import {
    Card, CardContent, CardHeader, makeStyles, Typography,
} from "@material-ui/core";
import TalkTrack from "./TalkTrack/TalkTrack";
import TalkTrackSkeleton from "./TalkTrackSkeleton";

const useStyles = makeStyles(({spacing}) => ({
    card: {
        padding: spacing(2, 3.5, 0),
    },
}));

function TalkTracksWrapper({ talkTrackWorkflows, headerAction, activeTalkTrackID, isATalkTrackBeingFetched }) {
  const classes = useStyles();

  return (
    <Card>
      <CardHeader className={classes.card} title="Talk Tracks" action={headerAction} />
      <CardContent>

        { !Array.isArray(talkTrackWorkflows) &&
            ( <TalkTrackSkeleton/> )
        }

        { ( Array.isArray(talkTrackWorkflows) && !talkTrackWorkflows.length ) &&
            ( <Typography>There are no active talk tracks.</Typography> )
        }

        { Array.isArray(talkTrackWorkflows) &&
          (
              <TalkTrack
                isATalkTrackBeingFetched={isATalkTrackBeingFetched}
                talkTrackItems={talkTrackWorkflows}
                activeTalkTrackID={activeTalkTrackID}
              />
          )
        }
      </CardContent>
    </Card>
  );
}

TalkTracksWrapper.propTypes = {
  talkTrackWorkflows: PropTypes.arrayOf(PropTypes.string).isRequired,
  headerAction: PropTypes.node,
  isATalkTrackBeingFetched: PropTypes.bool.isRequired,
};

TalkTracksWrapper.defaultProps = {
  headerAction: undefined,
};

export default TalkTracksWrapper;
