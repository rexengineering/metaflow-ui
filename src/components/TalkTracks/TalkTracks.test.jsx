import React from "react";
import { render } from "@testing-library/react";
import TalkTracks from "./TalkTracks";
import mockData from "./mockData";
import {Provider} from "react-redux";
import getStore from "../../store";

describe("<TalkTrack />", () => {

  const defaultProps = {
    activeTalkTrackID: "buying-123",
    talkTrackItems: mockData,
    isATalkTrackBeingFetched: false,
  };

  const renderTalkTrackComponent = ({ activeTalkTrackID, talkTrackItems, isATalkTrackBeingFetched} = defaultProps) => (
      <Provider store={getStore({})}>
        <TalkTracks
            isATalkTrackBeingFetched={isATalkTrackBeingFetched}
            talkTrackWorkflows={talkTrackItems}
            activeTalkTrackID={activeTalkTrackID}
        />
      </Provider>
  );

  it("should render", () => {
    const { asFragment } = render(renderTalkTrackComponent());
    const initialRender = asFragment();

    expect(initialRender).not.toBeNull();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(renderTalkTrackComponent());
    const initialRender = asFragment();

    expect(initialRender).toMatchSnapshot();
  });

});
