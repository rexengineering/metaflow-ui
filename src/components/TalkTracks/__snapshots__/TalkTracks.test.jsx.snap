// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TalkTracks /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          Talk Tracks
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        There are no active talk tracks.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<TalkTracks /> should render a header action if passed 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-text"
  data-testid="header-action"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label"
  >
    Action
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;
